### PR TITLE
Add parsed modules to module lookup registry in context

### PR DESCRIPTION
**Pull request by hinshun**
- Breaking up #286 into smaller PRs

### CODE REVIEW

File: parser/ast/context.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	mu   sync.Mutex
+}
+
+func NewModules() *ModuleLookup {
+	return &ModuleLookup{
+		mods: make(map[string]*Module),
+	}
+}
+
+func (ml *ModuleLookup) Get(filename string) *Module {
```
Comment (coryb) [inline parser/ast/context.go:33]: This isn't used anywhere, right?

**Review by coryb — APPROVED** [review]

File: parser/ast/context.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	ml.mu.Lock()
```
Comment (aaronlehmann) [inline parser/ast/context.go:34]: Would it make sense to use a `RWLock` and use `RLock()` here? Or does it not really matter?

Alternatively, we could use `sync.Map`, though the docs say:

```
The Map type is specialized. Most code should use a plain Go map instead, with separate locking or coordination, for better type safety and to make it easier to maintain other invariants along with the map content.
```

**Review by aaronlehmann — APPROVED** [review]

File: parser/ast/context.go (hunk 1 above)
Comment (hinshun) [reply parser/ast/context.go:33]: It's not used atm, I just wanted the eventual review of the debugger to be easier.

File: parser/ast/context.go (hunk 2 above)
Comment (hinshun) [reply parser/ast/context.go:34]: I can switch to `sync.RWMutex` to keep things simple.

Comment (hinshun) [reply parser/ast/context.go]: Done.

**Pull request merged**